### PR TITLE
add custom prompt to open AI translation

### PR DESCRIPTION
**Pull request by WatheqAlshowaiter**
some times an injected user prompt will enhance the translation, like this app is financial or medial app, and so on

### CODE REVIEW

File: src/Translator/OpenAiTranslator.php [hunk 1]
```diff
@@ -114,4 +122,9 @@ public function isConfigured(): bool
 
         return $openAiConfig['api_key'] && $openAiConfig['model'];
     }
+
+    private function CustomPrompt(): string
```
Comment (negoziator) [inline src/Translator/OpenAiTranslator.php:126]: Method should be in camelcase `customPrompt()` 😄

File: config/translator.php [hunk 2]
```diff
@@ -31,6 +31,14 @@
...
             'api_key' => env('OPENAI_API_KEY'),
             'organization_id' => env('OPENAI_ORGANIZATION'),
+
+            /**
+             * Custom prompt to enhance translation quality.
+             * 
+             * Example:
+             * 'custom_prompt' => 'This application translates medical terms consistently throughout.'
+             */
+            'custom_prompt' => ''
```
Comment (Bottelet) [inline config/translator.php:41]: `prompt_extension` perhaps? custom_prompt I would assume it replaced the entire prompt.

**Comment by Bottelet** [comment]
Good idea overall

File: config/translator.php (hunk 2 above)
Comment (WatheqAlshowaiter) [reply config/translator.php:41]: yes, I think this is better

Comment (WatheqAlshowaiter) [reply config/translator.php:41]: or maybe additional_prompt?

Comment (Bottelet) [reply config/translator.php:41]: Definitely better could also consider `prompt_context`.

Comment (WatheqAlshowaiter) [reply config/translator.php]: I have added now, you can think about better name for this

File: src/Translator/OpenAiTranslator.php (hunk 1 above)
Comment (WatheqAlshowaiter) [reply src/Translator/OpenAiTranslator.php]: fixed :  ) @negoziator

**Comment by WatheqAlshowaiter** [comment]
I have fixed the naming

@Bottelet

**Comment by Bottelet** [comment]
Thanks for the PR @WatheqAlshowaiter

**Pull request merged**